framework/src/org/apache/cordova/engine/SystemWebViewEngine.java

```diff
@@ -326,6 +329,12 @@ public void setPaused(boolean value) {
     @Override
     public void destroy() {
         webView.chromeClient.destroyLastDialog();
+        nativeToJsMessageQueue.reset();
```

Comment:
Why are we resetting the nativeToJsMessageQueue on destroy?  We may want to send some events before we kill the WebView.